src/pages/WishlistItem.jsx

```diff
@@ -0,0 +1,479 @@
+import React, { useState, useEffect } from 'react';
+import { useParams, useNavigate } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import ResponsiveContainer from '../components/styled/ResponsiveContainer';
+import {
+  Box,
+  Button,
+  Typography,
+  IconButton,
+  CardMedia,
+  Stack,
+  TextField,
+  InputAdornment,
+  Input,
+} from '@mui/material';
+import EditIcon from '@mui/icons-material/Edit';
+import CloseIcon from '@mui/icons-material/Close';
+import CurrencyExchangeIcon from '@mui/icons-material/CurrencyExchange';
+import LoadingSpinner from '../components/UIElements/LoadingSpinner';
+import { Controller, useForm } from 'react-hook-form';
+import {
+  getSingleWishlistItem,
+  deleteWishlistItem,
+  updateWishlistItem,
+} from '../redux/wishlist/wishlistActions';
+import {
+  setSuccess,
+  setLoading,
+  setItemToDelete,
+} from '../redux/wishlist/wishlistSlice';
+import {
+  setBasketName,
+  setDescription,
+  setMoneyGoal,
+  setPhotoTag,
+} from '../redux/basket/createBasketSlice';
+
+import useModal from '../hooks/useModal';
+
+const WishlistItem = () => {
+  const [editMode, setEditMode] = useState(false);
+
+  const navigate = useNavigate();
+  const modal = useModal();
+
+  let { id } = useParams();
+
+  const {
+    success,
+    loading,
+    singleItemInfo: itemInfo,
+  } = useSelector(state => state.wishlist);
+
+  const dispatch = useDispatch();
+
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm();
+
+  useEffect(() => {
+    dispatch(setItemToDelete(null));
+    dispatch(setSuccess(false));
+  }, []);
+
+  useEffect(() => {
+    getItemInfo();
+  }, [dispatch]);
+
+  const getItemInfo = async () => {
+    await dispatch(getSingleWishlistItem({ id }));
+  };
+
+  const formatteDate = date => {
+    const formatte = date => {
+      return date < 10 ? '0' + date : date;
+    };
+
+    const year = date.getFullYear().toString().slice(2);
+    const month = formatte(date.getMonth() + 1);
+    const day = formatte(date.getDate());
+
+    return `${day}/${month}/${year}`;
+  };
+
+  const submitForm = async data => {
+    await dispatch(updateWishlistItem({ id, data }));
+    setEditMode(false);
+  };
+
+  const removeItem = async () => {
+    await dispatch(deleteWishlistItem({ id }));
+  };
+
+  const resetStateStatusAndRedirect = async () => {
+    await dispatch(setSuccess(false));
+    await dispatch(setLoading(true));
+    navigate('/wishlist');
+  };
+
+  useEffect(() => {
+    if (success) {
+      resetStateStatusAndRedirect();
+    }
+  }, [success]);
+
+  const handleTransformButton = async () => {
```

Comment:
Are you sure that async/await are required here, didn't it work without them?

Reply:
yup, you're totally right, I don't need them
thank you